board/freescale/mx6ullevk/mx6ullevk_siklu_hw.c

```diff
@@ -94,64 +94,48 @@ static void setup_iomux_siklu_cpld(void) {
 }
 
 
-/* Note: ./cmd/siklu/siklu_nfs_boot.c setup_bootargs() uses the
- * 'siklu_board_type' environment variable because this API
- * function is not available there. See also spl_get_siklu_board_id()
- * in board/freescale/mx6ullevk/mx6ullevk.c. */
+/* See also spl_get_siklu_board_id() in board/freescale/mx6ullevk/mx6ullevk.c. */
 SKL_BOARD_TYPE_E siklu_get_board_type(void)
 {
-	static u8 is_ready = 0;
-	static SKL_BOARD_TYPE_E board_type;
+	SKL_BOARD_TYPE_E board_type;
+	
+	uint32_t reg_val;
+	ulong reg_addr = 0x20A0000;
+	u8 val;
 
-	if (!is_ready)
-	{
-		uint32_t reg_val;
-		ulong reg_addr = 0x20A0000;
-		u8 val;
-
-		reg_val = readl((uint32_t*)reg_addr);
-		//printf("reg_val - 0x%x\n",reg_val);
-
-		val = ((reg_val & 1<<18)?(1):(0)) + ((reg_val & 1<<19)?(2):(0))
-			+ ((reg_val & 1<<20)?(4):(0))
-			+ ((reg_val & 1<<21)?(8):(0));
-
-		//printf("board_id - 0x%x\n",val);
-
-		switch (val)
-		{
-		case 0:
-			board_type = SKL_BOARD_TYPE_PCB195;
-			env_set("siklu_board_type", "PCB195");
-			break;
-		case 1:
-			board_type = SKL_BOARD_TYPE_PCB213;
-			env_set("siklu_board_type", "PCB213");
-			break;
-		case 2:
-			board_type = SKL_BOARD_TYPE_PCB217;
-			env_set("siklu_board_type", "PCB217");
-			break;
-		case 3:
-			board_type = SKL_BOARD_TYPE_PCB277;
-			env_set("siklu_board_type", "PCB277");
-			break;
-		case 4:
-			board_type = SKL_BOARD_TYPE_PCB295;
-			env_set("siklu_board_type", "PCB295");
-			break;
-		case 5:
-			board_type = SKL_BOARD_TYPE_PCB295_AES;
-			env_set("siklu_board_type", "PCB295_AES");
-			break;
-		default:
-			board_type = SKL_BOARD_TYPE_UNKNOWN;
-			env_set("siklu_board_type", "unknown");
-			printf("Error: Unknown board type 0x%x\n", val);
-			break;
-		}
+	reg_val = readl((uint32_t*)reg_addr);
+	//printf("reg_val - 0x%x\n",reg_val);
+
+	val = ((reg_val & 1<<18)?(1):(0)) + ((reg_val & 1<<19)?(2):(0))
+		+ ((reg_val & 1<<20)?(4):(0))
+		+ ((reg_val & 1<<21)?(8):(0));
```

Comment:
```
val = (reg_val >> 17) & 0x1F;
```

Reply:
no, it constructs the value of `val` bit by bit depending on which bits are set in the `reg_val`
In any case, I did not change this line, this is how it was before (I just changed the indentation of the existing code)